monkey/infection_monkey/utils/signal_handler.py

```diff
@@ -1,50 +1,77 @@
 import logging
 import signal
+from typing import Optional
 
 from infection_monkey.i_master import IMaster
 from infection_monkey.utils.environment import is_windows_os
 
 logger = logging.getLogger(__name__)
 
 
+_signal_handler = None
+
+
 class StopSignalHandler:
     def __init__(self, master: IMaster):
         self._master = master
 
-    def handle_posix_signals(self, signum: int, _):
-        self._handle_signal(signum, False)
+    # Windows won't let us correctly deregister a method, but Callables and closures work.
+    def __call__(self, signum: int, *args) -> Optional[bool]:
+        if is_windows_os():
+            return self._handle_windows_signals(signum)
+        else:
+            self._handle_posix_signals(signum, args)
```

Comment:
Continuing https://github.com/guardicore/monkey/pull/1692/files#r798271812, why is one returned while the other isn't?

Also, if it's a reference to the function that is being returned, the return type hint should be changed.

Reply:
> Continuing https://github.com/guardicore/monkey/pull/1692/files#r798271812, why is one returned while the other isn't?

On Windows, signal handlers return a boolean. On Linux, signal handlers do not return anything.

> Also, if it's a reference to the function that is being returned, the return type hint should be changed.

Read about callables [[1](https://treyhunner.com/2019/04/is-it-a-class-or-a-function-its-a-callable/)][[2](https://pythonsimplified.com/what-is-a-callable-in-python/)]. If you still have questions we can discuss over zoom.

Reply:
Oh, I was just confused how `__call__()` worked. Got it now, thanks.